src/server/pipeline.py

```diff
@@ -232,9 +244,30 @@ async def perform_handshake(
                 f" established"
             )
         )
+        # Disconnect feeler connections and update connection timestamp.
+        if connection.connection_type == NodeType.FULL_NODE:
```

Comment:
Same issue here, doesn't belong in pipeline. Maybe you need visibility to the `is_feeler` variable? Seems like that's a part of `ChiaConnection`, so should be visible even on the other side of `_state_changed`.

Reply:
The full node object should not deal with any networking stuff, that is why the request/respond peers API call was put in the pipeline instead. 
We could factor this code out into a different file. 
I'm not sure state_changed_callback is the right place to put it. First of all, receiving a message does not trigger a state change. only new/removed connections.
Secondly, the RPC server class is the one that is getting the state updates (creates the state_changed callback). So it doesn't make much sense for the RPC server to deal with this stuff

Reply:
@mariano54 Not sure how the `_state_changed` infrastructure works... if it's tightly tied to the RPC only, maybe it could be made more general pub/sub so this connection manager code can get notifications too.